TST use global_random_seed in sklearn/decomposition/tests/test_kernel_pca.py

pedro9olivares opened the pull request:
Hi!

I'm opening this PR in regards to [this issue](https://github.com/scikit-learn/scikit-learn/issues/22827).

Modified the following functions:

- test_kernel_pca
- test_kernel_pca_consistent_transform
- test_kernel_pca_deterministic_output
- test_kernel_pca_sparse
- test_kernel_pca_linear_kernel
- test_kernel_pca_n_components
- test_kernel_pca_precomputed
- test_kernel_pca_inverse_correct_gamma

Added an additional argument "global_random_seed" for the RandomState which was previously hardcoded.

pedro9olivares committed:
TST use global_random_seed in sklearn/decomposition/tests
/test_kernel_pca.py

ogrisel commented:
Please push an extra empty commit to this PR to trigger the new test with all admissible random seeds on all CI runners as documented in #28959.

pedro9olivares commented:
> Please push an extra empty commit to this PR to trigger the updated tests with all admissible random seeds on all CI runners as documented in #28959.

Will do!

pedro9olivares committed:
checking on all runners [all random seeds]
test_kernel_pca
test_kernel_pca_consistent_transform
test_kernel_pca_deterministic_output
test_kernel_pca_sparse
test_kernel_pca_linear_kernel
test_kernel_pca_n_components
test_kernel_pca_precomputed
test_kernel_pca_inverse_correct_gamma

yuanx749 commented on sklearn/decomposition/tests/test_kernel_pca.py:
```diff
@@ -181,13 +181,13 @@ def test_kernel_pca_linear_kernel(solver, n_features):
 
 
-def test_kernel_pca_n_components():
+def test_kernel_pca_n_components(global_random_seed):
```
Seems `global_random_seed` is not needed for this test, since it only tests the output has the correct shape depending on `n_components`.

OmarManzoor replied:
Yeah I think that makes sense.

OmarManzoor commented on sklearn/decomposition/tests/test_kernel_pca.py:
```diff
@@ -181,13 +181,13 @@ def test_kernel_pca_linear_kernel(solver, n_features):
     on the selected number of components.
     """
-    rng = np.random.RandomState(0)
+    rng = np.random.RandomState(global_random_seed)
```
```suggestion
    rng = np.random.RandomState(0)
```

OmarManzoor committed:
Update sklearn/decomposition/tests/test_kernel_pca.py

OmarManzoor committed:
Update sklearn/decomposition/tests/test_kernel_pca.py

OmarManzoor committed:
Merge branch 'main' into for_global_random_seed

OmarManzoor reviewed (approved):
LGTM. Thanks @pedro9olivares

ArturoAmorQ commented:
Congrats on your first merge in scikit-learn, @pedro9olivares!
